manifests/config.pp

```diff
@@ -39,11 +39,12 @@
   $extra_config_sections         = $::squid::extra_config_sections,
 ) inherits squid {
 
-  concat{$config:
-    ensure => present,
-    owner  => $config_user,
-    group  => $config_group,
-    mode   => '0640',
+  concat { $config:
+    ensure       => present,
+    owner        => $config_user,
+    group        => $config_group,
+    mode         => '0640',
+    validate_cmd => '/usr/sbin/squid -k parse -f %',
```

Comment:
Do we hard code '/usr/sbin/squid' anywhere already ? Maybe this this should be an parameter? And if so that can be tested. I have no idea if squid is located on all systems in the same place? I doubt it.

Reply:
Would you prefer something like a `binary_path` parameter that would get /usr/sbin/squid or otherwise for different distros or something like `validate_cmd` which a user could then override ?

Reply:
> Do we hard code '/usr/sbin/squid' anywhere already ? Maybe this this should be an parameter? And if so that can be tested. I have no idea if squid is located on all systems in the same place? I doubt it.

Presumably the acceptance tests are at least verifying it for Debian, Ubuntu and RHEL.  I wouldn't be surprised if FreeBSD were different though.

Reply:
> Presumably the acceptance tests are at least verifying it for Debian, Ubuntu and RHEL. I wouldn't be surprised if FreeBSD were different though.

I think with my current code we would have trouble on Debian 8 where the path is `/usr/sbin/squid3`.

For me we have the following options:

- create a `binary_path` parameter that would point to it
- create a `validate_cmd` which would hold the full command

Reply:
+1 for the first option.  It'd need to go in [here](https://github.com/voxpupuli/puppet-squid/blob/master/manifests/params.pp#L44) somewhere.

`bin_path` or `squid_path`? (Naming things is *hard*) :)

Reply:
> +1 for the first option. It'd need to go in [here](https://github.com/voxpupuli/puppet-squid/blob/master/manifests/params.pp#L44) somewhere.
> 
> `bin_path` or `squid_path`? (Naming things is _hard_) :)

I'll go with `bin_path`

Reply:
`squid_bin_path`? :) On a more serious note, `bin_path` can be interpreted as `$PATH`.

Reply:
No problem.  Thanks for the PR!